substrate/frame/ranked-collective/src/lib.rs

```diff
@@ -650,6 +656,44 @@ pub mod pallet {
 				pays_fee: Pays::No,
 			})
 		}
+
+		/// Exchanges a member with a new account and the same existing rank.
+		///
+		/// - `origin`: Must be the `AdminOrigin`.
+		/// - `who`: Account of existing member of rank greater than zero to be exchanged.
+		/// - `new_who`: New Account of existing member of rank greater than zero to exchanged to.
+		/// - `min_rank`: The rank of the member or greater.
+		///
+		/// Weight: `O(min_rank)`.
+		#[pallet::call_index(6)]
+		#[pallet::weight(T::WeightInfo::exchange_member())]
+		pub fn exchange_member(
+			origin: OriginFor<T>,
+			who: AccountIdLookupOf<T>,
+			new_who: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			// remove who
+			let max_rank = T::ExchangeOrigin::ensure_origin(origin)?;
+			let who = T::Lookup::lookup(who)?;
+			let MemberRecord { rank, .. } = Self::ensure_member(&who)?;
+
+			for r in 0..=rank {
+				Self::remove_from_rank(&who, r)?;
+			}
+			Members::<T, I>::remove(&who);
+
+			// add new member
+			let new_who = T::Lookup::lookup(new_who)?;
+			Self::do_add_member(new_who.clone())?;
+
+			// promotes new_who to the rank of who
+			for _i in 1..=rank {
+				let _ = Self::do_promote_member(new_who.clone(), Some(max_rank))?;
+			}
+
+			Self::deposit_event(Event::MemberExchanged { who, new_who });
```

Comment:
This event should probably be emitted after the before calling `do_promote_member`, since that will emit `RankChanged` events.

Reply:
I have now done what you suggested to use `do_add_member_to_rank` in the previous comment